File: app/src/androidTest/java/com/xmartlabs/template/ui/common/ViewActions.java
package com.xmartlabs.template.ui.common;

import android.support.test.espresso.UiController;
import android.support.test.espresso.ViewAction;
import android.view.View;

import org.hamcrest.Matcher;

/**
 * Created by medina on 21/09/2016.
 */
public class ViewActions {
  public static ViewAction nestedScrollViewScrollTo() {
    return android.support.test.espresso.action.ViewActions.actionWithAssertions(
        new ScrollToActionForNestedScrollView());
  }

  // http://stackoverflow.com/a/33516360/5170805
  public static ViewAction withCustomConstraints(final ViewAction action, final Matcher<View> constraints) {
    return new ViewAction() {
      @Override
      public Matcher<View> getConstraints() {
        return constraints;
      }

      @Override
      public String getDescription() {
        return action.getDescription();
      }

      @Override
      public void perform(UiController uiController, View view) {
        action.perform(uiController, view);
      }
    };
  }
}

File: app/src/androidTest/java/com/xmartlabs/template/ui/common/ViewActions.kt
package com.xmartlabs.template.ui.common

import android.support.test.espresso.UiController
import android.support.test.espresso.ViewAction
import android.view.View

import org.hamcrest.Matcher

import android.support.test.espresso.action.ViewActions as SupportViewActions

object ViewActions {
  fun nestedScrollViewScrollTo() = SupportViewActions.actionWithAssertions(ScrollToActionForNestedScrollView())

  // http://stackoverflow.com/a/33516360/5170805
  fun withCustomConstraints(action: ViewAction, constraints: Matcher<View>) = object : ViewAction {
    override fun getConstraints() = constraints
  
    override fun getDescription() = action.description
  
    override fun perform(uiController: UiController, view: View) = action.perform(uiController, view)
  }
}
